pkg/scheduler/cache/util.go

```diff
@@ -29,11 +32,47 @@ func CreateNodeNameToInfoMap(pods []*v1.Pod, nodes []*v1.Node) map[string]*NodeI
 		}
 		nodeNameToInfo[nodeName].AddPod(pod)
 	}
+	imageExistenceMap := createImageExistenceMap(nodes)
+
 	for _, node := range nodes {
 		if _, ok := nodeNameToInfo[node.Name]; !ok {
 			nodeNameToInfo[node.Name] = NewNodeInfo()
 		}
-		nodeNameToInfo[node.Name].SetNode(node)
+		nodeInfo := nodeNameToInfo[node.Name]
+		nodeInfo.SetNode(node)
+		nodeInfo.imageStates = getNodeImageStates(node, imageExistenceMap)
 	}
 	return nodeNameToInfo
 }
+
+// getNodeImageStates returns the given node's image states based on the given imageExistence map.
+func getNodeImageStates(node *v1.Node, imageExistenceMap map[string]sets.String) map[string]*ImageStateSummary {
```

Comment:
Could you please write unit tests for these two new functions?

Reply:
Done - please let me know if further unit tests are needed (e.g., for CreateNodeNameToInfoMap and in cache_test.go) as I am still learning/observing when unit tests are needed (I thought for public/exposed methods they are must-have..) :D - thank you Bobby!

Reply:
We should do our best to cover all of our code by unit and integration tests. I think we didn't have enough coverage in our existing tests for these functions. As you know, there was a bug in the next function.